### DISCUSSION TITLE
When to use .Include() if using ProjectTo()

### DISCUSSION THREAD

**Discussion by urielzen** [LuckyPennySoftware/AutoMapper#3713] [NA]
I am using if statements below to conditionally include `WorkHistory` and `ProfileSkills` entities. But I just noticed that regardless of the if statement result `true/false` the generated SQL query is always the same.

```csharp
 IQueryable<Data.Models.Profile> profileQuery = _db.ActiveProfiles
    .Include(p => p.User)
    .Where(p => p.UserId == request.UserId)
    .Include(p => p.AlternateAddress)
    .Include(p => p.CurrentLocation)
    .Include(p => p.HomeAddress)
    .Include(p => p.WorkAddress)
    .Include(p => p.Image);

if (request.In.IncludeWorkHistory)
{
    profileQuery = profileQuery.Include(p => p.WorkHistories.Where(i => i.DeletedAt == null));
}

if (request.In.IncludeProfileSkills)
{
    profileQuery = profileQuery.Include(p => p.ProfileSkills.Where(i => i.DeletedAt == null));
}

var profile = await profileQuery
    .ProjectTo<GetProfileOut>(mapperConfig)
    .SingleOrDefaultAsync(cancellationToken);
```

I am just getting myself familiarized with automapper, but my guess is that the ProjectTo() overrides the Includes? So I really need to use my if statements in the mapperConfig like so?

```csharp

var profileMapConfig = cfg.CreateMap<Data.Models.Profile, GetProfileOut>();

if (request.In.IncludeWorkHistory)
{
    profileMapConfig = profileMapConfig
        .ForMember(dest => dest.WorkHistory, opt => opt.MapFrom(src => src.WorkHistories));
}
else
{
    profileMapConfig = profileMapConfig.ForMember(dest => dest.WorkHistory, opt => opt.Ignore());
}

if (request.In.IncludeProfileSkills)
{
    profileMapConfig = profileMapConfig
        .ForMember(dest => dest.ProfileSkills, opt => opt.MapFrom(src => src.ProfileSkills));
}
else
{
    profileMapConfig = profileMapConfig.ForMember(dest => dest.ProfileSkills, opt => opt.Ignore());
}

```

Thanks!

**Comment by lbargaoanu** [comment]
Never :) You don't need `Include` with `ProjectTo`. What's in the DTO gets fetched, unless you use `ExplicitExpansion`.

**Reply by urielzen** [reply]
Thanks, now I have to figure out how to add to my mappingConfig the `Where(i => i.DeletedAt == null)` that I had in my Includes. 😄

**Reply by lbargaoanu** [reply]
`.ForMember(dest => dest.WorkHistory, opt => opt.MapFrom(src => src.WorkHistories.Where(i => i.DeletedAt == null)))`

**Reply by urielzen** [reply]
Thanks, that makes sense, I think there is a typo and the parenthesis between `WorkHistories).Where` shouldn't be there.

But thanks a lot for the help, I was trying some other crazy things I never realized the solution was the simpler approach.
